### PR TITLE
Sudo utility for establishing an HRMP channel

### PR DESCRIPTION
**Pull request by pepyakin**

### CODE REVIEW

File: runtime/common/src/paras_sudo_wrapper.rs [hunk 1]
```diff
@@ -88,5 +88,29 @@ decl_module! {
...
+		///
+		/// This is equivalent to sending an `Hrmp::hrmp_init_open_channel` extrinsic followed by
+		/// `Hrmp::hrmp_accept_open_channel`.
+		#[weight = (1_000, DispatchClass::Operational)]
+		pub fn sudo_establish_hrmp_channel(
+			origin,
+			sender: ParaId,
+			recipient: ParaId,
+			max_capacity: u32,
+			max_message_size: u32
```
Comment (bkchr) [inline runtime/common/src/paras_sudo_wrapper.rs:102]:
```suggestion
			max_message_size: u32,
```

File: runtime/common/src/paras_sudo_wrapper.rs [hunk 2]
```diff
@@ -88,5 +88,29 @@ decl_module! {
...
+			ensure_root(origin)?;
+
+			<hrmp::Module<T>>::init_open_channel(
+				sender,
+				recipient,
+				max_capacity,
+				max_message_size,
+			)?;
+			<hrmp::Module<T>>::accept_open_channel(recipient, sender)?;
+			Ok(())
```
Comment (bkchr) [inline runtime/common/src/paras_sudo_wrapper.rs:113]:
```suggestion
			<hrmp::Module<T>>::accept_open_channel(recipient, sender)
```

File: runtime/parachains/src/hrmp.rs [hunk 3]
```diff
@@ -808,7 +825,12 @@ impl<T: Config> Module<T> {
 		weight
 	}
 
-	pub(super) fn init_open_channel(
+	/// Initiate opening a channel from a parachain to a given recipient with given channel
+	/// parameters.
+	///
+	/// Basically the same as [`hrmp_init_open_channel`] but intendend for calling directly from
```
Comment (bkchr) [inline runtime/parachains/src/hrmp.rs:831]:
```suggestion
	/// Basically the same as [`hrmp_init_open_channel`](Module::hrmp_init_open_channel) but intendend for calling directly from
```

File: runtime/parachains/src/hrmp.rs [hunk 4]
```diff
@@ -902,7 +924,11 @@ impl<T: Config> Module<T> {
 		Ok(())
 	}
 
-	pub(super) fn accept_open_channel(origin: ParaId, sender: ParaId) -> Result<(), Error<T>> {
+	/// Accept a pending open channel request from the given sender.
+	///
+	/// Basically the same as [`hrmp_accept_open_channel`] but intendend for calling directly from
```
Comment (bkchr) [inline runtime/parachains/src/hrmp.rs:929]:
```suggestion
	/// Basically the same as [`hrmp_accept_open_channel`](Module::hrmp_accept_open_channel) but intendend for calling directly from
```

**Review by bkchr — APPROVED** [review]

File: runtime/common/src/paras_sudo_wrapper.rs (hunk 2 above)
Comment (pepyakin) [reply runtime/common/src/paras_sudo_wrapper.rs:113]: That's a pretty much intended, and I would prefer to keep it this way.

**Comment by pepyakin** [comment]
bot merge

**Pull request merged**